### PR TITLE
Cancel schedule update API

### PR DESCRIPTION
**Pull request by ketakig12**
This PR:
- Adds a new DELETE /schedule_update API
- The API deletes all update jobs (repeat job and one time retry jobs) for the instance 
- Scheduler and ScheduleManager cancel/delete job functions are made parameterized to accept whether to cancel all jobs or only a particular repeat job
- Unit tests added for the same
- API documentation changes #553

### CODE REVIEW

**Comment by kamath-prasad** [comment]
pls update labels once ready for review.

File: api-controllers/ServiceFabrikApiController.js [hunk 1]
```diff
@@ -1026,6 +1026,22 @@ class ServiceFabrikApiController extends FabrikBaseController {
         .send(body));
   }
 
+  cancelScheduledUpdate(req, res) {
+    if (!_.get(req, 'cloudControllerScopes').includes('cloud_controller.admin')) {
+      throw new Forbidden(`Permission denined. Cancelling of scheduled updates can only be done by user with cloud_controller.admin scope.`);
```
Comment (kamath-prasad) [inline api-controllers/ServiceFabrikApiController.js]: typo - `denied`

Comment (ketakig12) [reply api-controllers/ServiceFabrikApiController.js]: Done.

**Comment by ketakig12** [comment]
@kamath-prasad There are unit tests for cancel backup schedule function, which are passing. Also tested manually the same and it works fine.

File: api-controllers/ServiceFabrikApiController.js [hunk 2]
```diff
@@ -1026,6 +1026,22 @@ class ServiceFabrikApiController extends FabrikBaseController {
...
+    // Cancel repeat and all onetime jobs as well
+    // e.g. cancel both jobs 6dee3dd8-c990-40d8-93b7-efcaa2637c5e_ServiceInstanceAutoUpdate and 6dee3dd8-c990-40d8-93b7-efcaa2637c5e_ServiceInstanceAutoUpdate_30minutesfromnow_1543823280862
+    const cancelAllJobs = true;
+    return Promise
+      .try(() => this.setPlan(req))
+      .then(() => ScheduleManager
+        .cancelSchedule(req.params.instance_id, CONST.JOB.SERVICE_INSTANCE_UPDATE, cancelAllJobs))
+      .then(() => res
+        .status(CONST.HTTP_STATUS_CODE.OK)
+        .send({}));
```
Comment (jagadish-kb) [inline api-controllers/ServiceFabrikApiController.js]: can you add a logger here with tap.. saying `Schedule update cancelled for ${instace_id} by ${req.user}`

**Comment by ketakig12** [comment]
@jagadish-kb Sorry I missed this, thanks for commenting! I made required changes.

**Comment by kamath-prasad** [comment]
As review comments are incorporated , going ahead and merging the PR.

**Pull request merged**